test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -315,14 +300,16 @@
            (is (true? (set-constants-and-verify pool-context 3)))
 
            ;; The flush should still not have completed at this point because
-           ;; the last instance from the old pool is still being borrowed
-           (is (not (realized? flush-complete)))
+           ;; the last instance from the old pool is still being borrowed.
+           ;; Wait for some arbitrary (but short) period of time to see that
+           ;; that the agent is still busy - presumably handling the flush.
+           (is (false? (await-for 50 pool-agent)))
```

Comment:
This seems maybe kind of racy? Although I guess if it passes locally there's probably not much chance the flush would be faster (and therefore fail) in Travis?